### ISSUE TITLE
Require global response timeouts configuration via helm values

### ISSUE THREAD

**Issue by Nbarola** [linkerd/linkerd2#10056]
### What problem are you trying to solve?

Rather than configuring timeouts for each [routes](https://linkerd.io/2.12/tasks/configuring-timeouts/#:~:text=Each%20route%20may%20define%20a,default%20timeout%20is%2010%20seconds), wanted a global configurable value to set global response timeouts for proxies.

### How should the problem be solved?

The document mentions that if unspecified, the default response timeout is 10 seconds, so we can have a global response timeout for all routes/proxies.

It would be great if we have something similar in helm values as we have for `outboundConnectTimeout` for establishing outbound connection

```
proxy:
  # -- Maximum time allowed for the proxy to establish an outbound TCP
  # connection
  outboundConnectTimeout: 1000ms
  # -- Maximum time allowed for the proxy to establish an inbound TCP
  # connection
  inboundConnectTimeout: 100ms
  ```
  It can be like
  ```
  proxy:
     responseTimeout: 10000ms
  ```
  
  

### Any alternatives you've considered?

Adding timeouts to all the routes, but we have to configure all the routes first in order to do that.

### How would users interact with this feature?

This value can be configured in helm custom values manifest file and can be used while installing/upgrading linked control plane via helm

### Would you like to work on this feature?

None

**Comment by kleimkuhler** [comment]
This type of configuration is something that will be handled via [Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/). Linkerd is working towards adopting the Gateway API which has been the foundation for the previous and upcoming policy work.

With policy attachment, things like global timeouts on routes can be attached at any level that you'd like — from a namespace or an individual workload. Additionally, this will keep the configuration out of Helm and other install-time values.

I'd keep an eye on the ongoing work which is working towards a greater adoption of the Gateway API and at some point this should be achievable with the properly supported resources.

**Comment by dwilliams782** [comment]
Is the default response timeout of 10 seconds still valid? I'm trying to chase down some 504s in our environments that have service profile routes defined, but often report much higher than 10s without changing the default config at all.

**Comment by kleimkuhler** [comment]
Hey sorry to just be getting back to you. Yep without additional configuration the response timeout should still be 10 seconds. If you're seeing issues with that not being the case, happy to look through things more in another issue if you have time to open one up with more details!

**Comment by dwilliams782** [comment]
Hi mate. I realised since that 504s from linkerd timeouts don't surface as 504s (I have an old issue about this [here](https://github.com/linkerd/linkerd2/issues/8915)), so it's unrelated.

I've never managed to prove the 10s timeout though - even creating a dummy nginx server which sleeps 15/30 seconds and sends a request works perfectly:

![image](https://user-images.githubusercontent.com/42545485/216294300-f3d5190c-debb-4296-b555-ff49ae2a2234.png)

I did this a long time ago so I no longer have the test manifests but it will still stand, and this service profile had no additional timeouts configured.
